services/api/src/models/Quests.test.ts

```diff
@@ -65,6 +66,49 @@ describe('quest', () => {
         .catch(done.fail);
     });
 
+    test('return private quests (alongwith public quests) when showPrivate is set to true', (done: DoneFn) => {
+      testingDBWithState(quests)
+        .then((tdb) => {
+          return searchQuests(tdb, '', {partition: PUBLIC_PARTITION, showPrivate: true});
+        })
+        .then((results) => {
+          console.log(results);
+          expect(results.length).toEqual(2);
+          done();
+        })
+        .catch(done.fail);
+    });
+
+
+    test('return private quests at the top (above public quests) when showPrivate is set to true', (done: DoneFn) => {
+      testingDBWithState(quests)
+        .then((tdb) => {
+          return searchQuests(tdb, '', {partition: PUBLIC_PARTITION, showPrivate: true});
+        })
+        .then((results) => {
+          expect(results.length).toEqual(2);
+          expect(results[0].partition).toEqual(PRIVATE_PARTITION);
+          expect(results[1].partition).toEqual(PUBLIC_PARTITION);
+          done();
+        })
+        .catch(done.fail);
+    });
+
+    test('doesn\'t return private quests when showPrivate is set to false', (done: DoneFn) => {
+      testingDBWithState(quests)
+        .then((tdb) => {
+          return searchQuests(tdb, '', {partition: PUBLIC_PARTITION, showPrivate: false});
+        })
+        .then((results) => {
+          expect(results.length).toEqual(1);
+          expect(results[0].partition).toEqual(PUBLIC_PARTITION);
```

Comment:
Similar thought here; rather than checking for length and partition of one quest, we can make this more robust by just confirming that all results have partition = public

Reply:
Looks like this still needs to be updated to be more generic - iterate through all results and expect each result's partition to equal public